compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -263,7 +263,13 @@ impl<'tcx> fmt::Display for FrameInfo<'tcx> {
             }
             if !self.span.is_dummy() {
                 let lo = tcx.sess.source_map().lookup_char_pos(self.span.lo());
-                write!(f, " at {}:{}:{}", lo.file.name, lo.line, lo.col.to_usize() + 1)?;
+                write!(
+                    f,
+                    " at {}:{}:{}",
+                    lo.file.name.prefer_remapped(),
+                    lo.line,
+                    lo.col.to_usize() + 1
+                )?;
```

Comment:
@rust-lang/miri, what would you like the behavior of this method to be in case `--remap-path-prefix` affects a filename here? Should the method use the remapped path (`lo.file.name.prefer_remapped()`) or should it still use the local file system path (`lo.file.name.prefer_local()`)? 

If the result of this method ever ends up in a library or binary or something else that gets shipped then this must use `prefer_remapped()` in order to enable preserving privacy and reproducible builds.

Reply:
I think this method is only used to format stack traces for error messages. What do we do for other rustc diagnostics?

Reply:
Other places are likely using `SourceMap::span_to_string()`. https://github.com/rust-lang/rust/blob/23f769ec7776521a5cc393219d1558bb3d83f686/compiler/rustc_span/src/source_map.rs#L409-L424

Reply:
It is possible we didn't use that since we only wanted the `lo` to be printed...
but anyway if that uses `prefer_remapped`, this should do the same. Maybe add a comment that this should be kept consistent with `SourceMap::span_to_string`=

Reply:
If it's only for diagnostics and will not end up in binary it probably should use `prefer_local()`

@michaelwoerister Do you think we should add a `prefer_local` argument to `span_to_string` as well?

Reply:
I changed this to `prefer_local()` because we want that in diagnostics. It broke a lot of tests with `remap-debuginfo = true` otherwise

Reply:
If this is used only for error messages then `prefer_local()` is the way to go. Thanks, @RalfJung!